compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -522,6 +522,9 @@ impl<'ll> StaticMethods for CodegenCx<'ll, '_> {
             }
 
             if attrs.flags.contains(CodegenFnAttrFlags::USED) {
+                // `USED` and `USED_LINKER` can't be used together.
+                assert!(!attrs.flags.contains(CodegenFnAttrFlags::USED_LINKER));
```

Comment:
It would be good to test that `#[used(compiler)] #[used(linker)] static FOO: u8 = 0;` errors out earlier and doesn't hit this assertion failure.